### PR TITLE
allow for input of comma in amount button, and display this value in …

### PR DESCRIPTION
**Pull request by ScottDowne**
…the proper locale if it came from query string

### CODE REVIEW

**Comment by ScottDowne** [comment]
WIP

**Comment by ScottDowne** [comment]
Needs tests.

**Comment by ScottDowne** [comment]
Added tests and I think karma should pass. Waiting for travis to go green.

@alicoding can I get a really thorough review?

File: components/amount-buttons.jsx [hunk 1]
```diff
@@ -53,7 +60,36 @@ var AmountOtherButton = React.createClass({
       this.props.onRadioChange();
     }
   },
+  onInputChange: function(e) {
+    var inputValue = e.currentTarget.value;
+    var amount = "";
+
+    if (/^(\d)*[\,]?(\d){0,2}$/.test(inputValue)) {
```
Comment (alicoding) [inline components/amount-buttons.jsx]: Can you add a comment above this regex that you are trying to capture as many number follow by comma and follow by 0 to 2 numbers?

File: components/amount-buttons.jsx [hunk 2]
```diff
@@ -53,7 +60,36 @@ var AmountOtherButton = React.createClass({
...
+      amount = inputValue.replace(",", ".");
+    } else if (/^(\d)*[\.]?(\d){0,2}$/.test(inputValue)) {
```
Comment (alicoding) [inline components/amount-buttons.jsx]: And here.

File: components/amount-buttons.jsx (hunk 1 above)
Comment (alicoding) [inline components/amount-buttons.jsx]: Actually I think this test is not enough. We need to consider about locales that uses comma as a separator like in the US and Canada e.g. 2,000.00 How are we going to deal with this?

Comment (alicoding) [inline components/amount-buttons.jsx]: My suggestion would be to check the currency/locale here as well.

File: components/amount-buttons.jsx (hunk 2 above)
Comment (alicoding) [inline components/amount-buttons.jsx]: Same goes for this case. 2.000,00 ?

File: components/amount-buttons.jsx [hunk 3]
```diff
@@ -53,7 +60,36 @@ var AmountOtherButton = React.createClass({
...
+  },
   render: function() {
+    var amount = this.props.amount;
+    var inputValue = this.state.inputValue;
+    if (!amount) {
+      inputValue = "";
+    } else if (!this.props.userInputting) {
+      // We only need this for initial display before the user starts inputting,
+      // once they start inputting, we can adapt.
+      inputValue = new Intl.NumberFormat(this.props.locale, {
```
Comment (alicoding) [inline components/amount-buttons.jsx]: Can't you use `this.formatNumber()` instead here?

**Comment by alicoding** [comment]
We need more test case for the case of 2,000.00 vs 2.000,00

File: components/amount-buttons.jsx (hunk 3 above)
Comment (ScottDowne) [inline components/amount-buttons.jsx]: Yes

File: components/amount-buttons.jsx (hunk 1 above)
Comment (ScottDowne) [inline components/amount-buttons.jsx]: It makes turning 1,00 -> 1.00 much harder. I need to start parsing and I kinda wanted to attempt a solution first that didn't parse.

Comment (ScottDowne) [inline components/amount-buttons.jsx]: Right now adding a dot or comma can only happen once, and only before the last two characters, making it really easy to parse.

Comment (alicoding) [inline components/amount-buttons.jsx]: does it switch based on the locale?

**Comment by ScottDowne** [comment]
@alicoding Updated, travis is running, should be green. R?

**Pull request merged**